GUIDELINES.md

```diff
@@ -207,6 +207,47 @@ Make sure all tests are passing with:
 
     $ cargo test --all-features
 
+### Running end-to-end tests
+
+In order to run end-to-end (e2e) tests you need to have a specific nightly toolchain.
+"Nightly" is necessary to use optimization compiler flags and have contract wasm small enough to be eligible for
+deployment.
+
+Run the following commands to install the necessary toolchain:
+
+```shell
+rustup install nightly-2024-01-01
+rustup component add rust-src
+```
+
+Also, you should have the cargo stylus tool:
+
+```shell
+cargo install cargo-stylus
+```
+
+Since most of the e2e tests using koba for deploying contracts, you need to install solidity compiler v0.8.21, like
+this:
+
+```shell
+curl -LO https://github.com/ethereum/solidity/releases/download/v0.8.21/solc-static-linux
+sudo mv solc-static-linux /usr/bin/solc
+sudo chmod a+x /usr/bin/solc
```

Comment:
Isn't it easier if people use npm? 
`npm install -g solc`

Reply:
wouldn't that require having `npm` installed, as opposed to `curl` (which I believe comes in every UNIX system)?

Reply:
I would also prefer smth without external deps. But we have settled on official solc installation guide.